publish/api.py

```diff
@@ -1,20 +1,30 @@
 from rest_framework import viewsets
 from rest_framework.decorators import action
 from rest_framework.pagination import PageNumberPagination
+from rest_framework.permissions import IsAuthenticated
 from rest_framework.response import Response
 
 from .models import Dandiset, NWBFile, Subject
 from .serializers import DandisetSerializer, DandisetPublishSerializer, NWBFileSerializer, SubjectSerializer
 from .tasks import publish_dandiset
 
 
+class PermissionMixin:
```

Comment:
Why not use [`IsAuthenticatedOrReadOnly`](https://www.django-rest-framework.org/api-guide/permissions/#isauthenticatedorreadonly)?

Reply:
I had not heard of it. I will include that in my next PR